common/flatpak-instance-private.h

```diff
@@ -58,4 +60,6 @@ gboolean flatpak_instance_ensure_per_app_xdg_runtime_dir (const char *app_id,
                                                           char **shared_dir_out,
                                                           GError **error);
 
+GStrv flatpak_instance_get_run_environ (FlatpakInstance *self);
```

Comment:
This should return a `const char * const *`, otherwise a reader might think that it returns a copy (`(transfer full)` or `(transfer container)` in GLib jargon). This will require a cast:

```
return (const char * const *) priv->run_environ;
```

but that's OK.

Reply:
It looks like we have this issue in `flatpak_instance_get_info` too, but it's public API.

Reply:
Actually, it seems that the `GKeyFile` case could be different; its own public API always expects non-`const` instances for some reason, even for seemingly read-only operations.

Reply:
GKeyFile is a reference-counted object, so it usually isn't useful to have non-const pointers to it, because you can't take or release a reference via a const pointer. The convention in GLib is that simple data types like `char **` are as const-correct as we can easily make them (which isn't 100% but is better than nothing), but more complicated refcounted objects are not const.

Reply:
Right, I guess it makes sense.

Anyway, `const char * const *` is used wherever it makes sense in the context of `run_environ`.